packages/pigeon/bin/run_tests.dart

```diff
@@ -121,6 +121,29 @@ Future<int> _runFlutterUnitTests() async {
     return generateCode;
   }
 
+  final int generateTestCode = await _runPigeon(
+    input: 'pigeons/message.dart',
+    dartOut: '$flutterUnitTestsPath/lib/message.gen.dart',
+    dartTestOut: '$flutterUnitTestsPath/test/message_test.dart',
+    streamOutput: true,
+  );
+  if (generateTestCode != 0) {
+    return generateTestCode;
+  }
+
+  final int analyzeCode = await _runProcess(
+    'flutter',
+    <String>['analyze'],
+    workingDirectory: flutterUnitTestsPath,
+  );
+  if (analyzeCode != 0) {
+    return analyzeCode;
+  }
+
+  // Delete these files that were just generated to help with the analyzer step.
+  File('$flutterUnitTestsPath/lib/message.gen.dart').deleteSync();
+  File('$flutterUnitTestsPath/test/message_test.dart').deleteSync();
+
```

Comment:
Also, this whole new block seems like it would be much better as a separate method, so that it's clear that it's a self-contained block that's unrelated to the code above and below (e.g., `Future<int> _analyzeGeneratedCode(String path)`). That would make this method easier to follow since it would be higher level..

Reply:
done